base/loading.jl

```diff
@@ -2,6 +2,52 @@
 
 # Base.require is the implementation for the `import` statement
 
+# Generic case-sensitive version of isfile
+function isfile_casesensitive_slow(path)
+    isfile(path) || return false
+    dir, filename = splitdir(path)
+    any(readdir(dir) .== filename)
+end
+
+isfile_casesensitive(path) = isfile_casesensitive_slow(path)
+
+@windows_only function isfile_casesensitive(path)
+    isfile(path) || return false
+    longpath(path) == path
+end
+
+@linux_only isfile_casesensitive(path) = isfile(path)
+
+const ATTR_BIT_MAP_COUNT=5
+const ATTR_CMN_NAME=1
+const BITMAPCOUNT = 1
+const COMMONATTR = 5
+const FSOPT_NOFOLLOW=1
+
+const attr_list = zeros(UInt8, 24)
+attr_list[BITMAPCOUNT] = ATTR_BIT_MAP_COUNT
+attr_list[COMMONATTR] = ATTR_CMN_NAME
+
+@osx_only function isfile_casesensitive(path)
+    isfile(path) || return false
+    buf_size = length(path) + 12
+    local canon_path
+    while true
+        buf = zeros(UInt8, buf_size)
+        ret = ccall(:getattrlist, Cint, (Cstring, Ptr{Void}, Ptr{Void}, Csize_t, Culong), path,
+          attr_list, buf, sizeof(buf), FSOPT_NOFOLLOW)
+        ret == 0 || error(Libc.errno())
+        len = unsafe_load(convert(Ptr{UInt32}, pointer(buf)), 3)
+        if (len + 12) >= buf_size
+            buf_size *= 2
+            continue
+        end
+        canon_path = bytestring(pointer(buf, 13))
```

Comment:
Couldn't you just do `canon_path = UTF8String(buf)` in order to avoid making a copy?